outputs.tf

```diff
@@ -6,6 +6,10 @@ output "bucket_domain_name" {
 output "bucket_id" {
   value       = module.s3_bucket.bucket_id
   description = "Bucket ID"
+  depends_on = [
+    # The S3 bucket policy resource must be created before the bucket is referenced by CloudTrail
+    module.s3_bucket
+  ]
```

Comment:
@X-Guardian, I talked to Cloudposse Engs about this pattern, and they do not recommend using it.

Reply:
Did whoever you talked to actually look at the issue that this PR is resolving? This module serves only one purpose which is to create the S3 Bucket for the [terraform-aws-cloudtrail](https://github.com/cloudposse/terraform-aws-cloudtrail) module to consume. With the current versions of both modules, this fails on creation as described in #90. Adding the output `depends_on` is the simple (and only) way to resolve this.

Reply:
@nuru could comment on here.

Reply:
I am uncomfortable allowing the use of an undocumented feature, and this should probably be fixed upstream in [terraform-aws-s3-bucket](https://github.com/cloudposse/terraform-aws-s3-bucket), but it works with Terraform 1.3, 1.5, and 1.8 and I don't see a better way to achieve the desired result, so I will allow this and we can see how it goes.